main.go

```diff
@@ -131,42 +131,48 @@ func executeSequentially(ctx context.Context, logger *log.Entry, c client, alloc
 
 			output, err := allocationExec(ctx, c, allocInfo, cmd)
 			if err != nil {
-				return fmt.Errorf("failed to exec on allocation: %w", err)
+				return fmt.Errorf("failed to execute command: %w", err)
 			}
 
 			logger.Info("command executed")
 
-			if err := consumeExecOutput(output); err != nil {
+			exitCode, err := consumeExecOutput(output)
+			if err != nil {
 				return fmt.Errorf("failed to consume command output: %w", err)
 			}
+
+			if exitCode != 0 {
+				return fmt.Errorf("command failed with code %d", exitCode)
+			}
 		}
 	}
 
 	return nil
 }
 
-func executeConcurrently(ctx context.Context, logger *log.Entry, c client, allocsInfo []*allocInfo, cmd []string, concurrency int) error {
+func executeConcurrently(ctx context.Context, logger *log.Entry, c client, allocsInfo []*allocInfo, cmd []string, concurrency int) error { // nolint: funlen
 	nbAllocs := len(allocsInfo)
 
 	execOutputCh := make(chan *execOutput, nbAllocs)
 	done := make(chan bool, 1)
 
-	defer func() {
-		close(execOutputCh)
-		<-done
-	}()
-
-	go func() {
+	var exitCode int
+	go func(exitCode *int) {
 		defer func() {
 			done <- true
 		}()
 
-		for out := range execOutputCh {
-			if err := consumeExecOutput(out); err != nil {
+		for output := range execOutputCh {
+			c, err := consumeExecOutput(output)
+			if err != nil {
 				log.Errorf("failed to consume command output: %v", err)
 			}
+
+			if c != 0 {
+				*exitCode = c
+			}
```

Comment:
You wouldn't have to modify `consumeExecOutput`

```suggestion
			if output.exitCode != 0 {
				*exitCode = output.exitCode
			}
```

Reply:
Oh thanks!